sbin/create_course.sh

```diff
@@ -14,56 +14,56 @@ fi
 
 CONF_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"/../config
 
-SUBMITTY_REPOSITORY_DIR=$(jq -r '.submitty_repository' ${CONF_DIR}/submitty.json)
-SUBMITTY_INSTALL_DIR=$(jq -r '.submitty_install_dir' ${CONF_DIR}/submitty.json)
-SUBMITTY_DATA_DIR=$(jq -r '.submitty_data_dir' ${CONF_DIR}/submitty.json)
-SUBMISSION_URL=$(jq -r '.submission_url' ${CONF_DIR}/submitty.json)
+SUBMITTY_REPOSITORY_DIR=$(jq -r '.submitty_repository' "${CONF_DIR}/submitty.json")
+SUBMITTY_INSTALL_DIR=$(jq -r '.submitty_install_dir' "${CONF_DIR}/submitty.json")
+SUBMITTY_DATA_DIR=$(jq -r '.submitty_data_dir' "${CONF_DIR}/submitty.json")
+SUBMISSION_URL=$(jq -r '.submission_url' "${CONF_DIR}/submitty.json")
 
-PHP_USER=$(jq -r '.php_user' ${CONF_DIR}/submitty_users.json)
-DAEMON_USER=$(jq -r '.daemon_user' ${CONF_DIR}/submitty_users.json)
-CGI_USER=$(jq -r '.cgi_user' ${CONF_DIR}/submitty_users.json)
+PHP_USER=$(jq -r '.php_user' "${CONF_DIR}/submitty_users.json")
+DAEMON_USER=$(jq -r '.daemon_user' "${CONF_DIR}/submitty_users.json")
+CGI_USER=$(jq -r '.cgi_user' "${CONF_DIR}/submitty_users.json")
 
-COURSE_BUILDERS_GROUP=$(jq -r '.course_builders_group' ${CONF_DIR}/submitty_users.json)
+COURSE_BUILDERS_GROUP=$(jq -r '.course_builders_group' "${CONF_DIR}/submitty_users.json")
 
-DATABASE_HOST=$(jq -r '.database_host' ${CONF_DIR}/database.json)
-DATABASE_PORT=$(jq -r '.database_port' ${CONF_DIR}/database.json)
-DATABASE_USER=$(jq -r '.database_user' ${CONF_DIR}/database.json)
-DATABASE_PASS=$(jq -r '.database_password' ${CONF_DIR}/database.json)
+DATABASE_HOST=$(jq -r '.database_host' "${CONF_DIR}/database.json")
+DATABASE_PORT=$(jq -r '.database_port' "${CONF_DIR}/database.json")
+DATABASE_USER=$(jq -r '.database_user' "${CONF_DIR}/database.json")
+DATABASE_PASS=$(jq -r '.database_password' "${CONF_DIR}/database.json")
 
 ########################################################################################################################
 ########################################################################################################################
 
 CONN_STRING="-h ${DATABASE_HOST} -U ${DATABASE_USER}"
-if [ -d ${DATABASE_HOST} ]; then
+if [ -d "${DATABASE_HOST}" ]; then
     CONN_EXTRA="${CONN_STRING} -p ${DATABASE_PORT}"
 fi
 
 # Check that Submitty Master DB exists.
 PGPASSWORD=${DATABASE_PASS} psql ${CONN_STRING} -lqt | cut -d \| -f 1 | grep -qw submitty
-if [[ $? -ne "0" ]] ; then
+if [[ "$?" -ne "0" ]] ; then
     echo "ERROR: Submitty master database doesn't exist."
     exit
 fi
 
 #Ensure that tables exist within Submitty Master DB.
 sql="SELECT count(*) FROM pg_tables WHERE schemaname='public' AND tablename IN ('terms','courses','courses_users','sessions','users');"
 table_count=`PGPASSWORD=${DATABASE_PASS} psql ${CONN_STRING} -d submitty -tAc "${sql}"`
```

Comment:
Consider using `$(...)` notation ([SC2006](https://www.shellcheck.net/wiki/SC2006)) and quoting the `${DATABASE_PASS}` ([SC2086](https://www.shellcheck.net/wiki/SC2086))?

```suggestion
table_count=$(PGPASSWORD="${DATABASE_PASS}" psql ${CONN_STRING} -d submitty -tAc "${sql}")
```

Reply:
If you see the previous commit, I actually had to unquote `DATABASE_PASS` since it's meant to be a number of CLI arguments.  It's a really really hacky solution, but I wanted to minimize the changes and I don't believe there are any security risks coming from it.  

As far as the SC2006 goes, I'm planning to go through and make another PR to fix other things like that.  There are a few tricky changes that are necessary in some cases (including for other errors) so I want to get the unquoted variables out of the way before beginning that PR.

Reply:
Yeah, I saw your last commit.  I think you are referring to `CONN_STRING` rather than `DATABASE_PASS`...  The definitions of the variables are:

- `DATABASE_PASS`: [L31](../blob/main/sbin/create_course.sh#L31)
  `DATABASE_PASS=$(jq -r '.database_password' "${CONF_DIR}/database.json")`
- `CONN_STRING`: [L36](../blob/main/sbin/create_course.sh#L36)
  `CONN_STRING="-h ${DATABASE_HOST} -U ${DATABASE_USER}"`

`CONN_STRING` does have some parameters that will pass into `psql`... Quoting it will definitively introduce problems.  

<hr/>
All SC2006, SC2115 related comments are marked as resolved.